integration-cli/docker_api_containers_test.go

```diff
@@ -1948,3 +1949,27 @@ func (s *DockerSuite) TestContainerKillCustomStopSignal(c *check.C) {
 	err = waitInspect(id, "{{.State.Running}} {{.State.Restarting}}", "false false", 30*time.Second)
 	c.Assert(err, checker.IsNil)
 }
+
+func (s *DockerSuite) TestCpAPIWithHardlink(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	containerID := makeTestContainer(c, testContainerOptions{addContent: true})
+
+	tmpDir := getTestDir(c, "test-cp-from-dir-with-hardlink")
+	defer os.RemoveAll(tmpDir)
+
+	srcPath := "/dir6"
+	dstPath := cpPath(tmpDir, "dir6_")
+
+	res, body, err := request.SockRequestRaw("GET", fmt.Sprintf("/containers/%s/archive?path=%s", containerID, srcPath), nil, "application/x-tar", daemonHost())
+	c.Assert(err, checker.IsNil)
+	c.Assert(res.StatusCode, checker.Equals, http.StatusOK)
+
+	srcInfo := archive.CopyInfo{
+		Path:   srcPath,
+		Exists: true,
+		IsDir:  true,
+	}
+	archive.CopyTo(body, srcInfo, dstPath)
```

Comment:
I don't think I understand why this is copying here. Shouldn't the test just assert that the file is where it is expected to be?  I think you could stat both and compare the `os.FileInfo` ?